src/android/CodePush.java

```diff
@@ -461,6 +487,8 @@ public void onStart() {
                 handleAppStart();
                 this.markUpdate();
                 this.codePushPackageManager.clearPendingInstall();
+            } else if (codePushReportingManager.hasFailedReport()) {
```

Comment:
On restart, notifyApplicationReady will be called anyway, and that will handle the reporting.